.github/workflows/markdown.yml

```diff
@@ -0,0 +1,36 @@
+# This is a basic workflow to help you get started with Actions
+name: Auto Format Documentation
+
+# Controls when the action will run. 
+on:
+  # Triggers the workflow on push or pull request events but only for the master branch
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# A workflow run is made up of one or more jobs that can run sequentially or in parallel
+jobs:
+  # This workflow contains a single job called "build"
+  format:
+    # The type of runner that the job will run on
+    runs-on: ubuntu-latest
+
+    # Steps represent a sequence of tasks that will be executed as part of the job
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        # Make sure the actual branch is checked out when running on pull requests
+        ref: ${{ github.head_ref }}
+
+    - name: Prettify code
+      uses: creyD/prettier_action@v3.3
+      with:
+        # This part is also where you can pass other options, for example:
+        prettier_options: --write **/*.md
```

Comment:
Does this change the files directly in the branch? It looks like it is, in which case I'm wondering if theres any point in having it on both PR and Push. Typically I've seen workflows be split into separate actions for PR / Push

Reply:
Yeah the on-push hook is not needed. The bot does not have permission to change master so it would just fail if there were still formatting issues after merging. Will make it for PR only